### ISSUE TITLE
Classification on CIFAR-10 Dataset

### ISSUE THREAD

**Issue by JessieSun-lalala** [XzwHan/CARD#7]
Hi, thanks for sharing. When I implemented classification task on CIFAR10 dataset, I encountered some problems.
First, I used the checkpoint you provided and got an accuracy of 94.2300%, which is higher than the 90.93% mentioned in the CARD paper.
Secondly, when I train CARD for classification on CIFAR10, the accuracy of the guidance classifier and generative models on the test set are both 0.94310000. 

I wonder if such a result is reasonable. Is there a mistake in my operation? Thank you for your help!

**Comment by XzwHan** [comment]
Hi @JiaxinSun0604, thank you for the question. We re-ran the code and found out that the setting of `trained_aux_cls_ckpt_name` argument in `cifar10.yml` file didn't match the one we applied in our paper; we have fixed it in the repo. Please give the updated code a try, and let us know if there are any issues. Thank you very much for catching this error from our end.

Specifically, we used the checkpoint `snapshot_supervised.ckpt` ($90.39\\%$ test accuracy) under the [`pretrained/cifar10_ckpt`](https://drive.google.com/drive/u/2/folders/17qP1rs4ENtLqwFcWAmbfRVX-dEalk1pD) directory as the $\boldsymbol{f_{\phi}}$ network to obtain the experiment results reported in our paper. The other checkpoint `snapshot_self_supervised.ckpt` gives $94.31\\%$ test accuracy. We mainly focused on `snapshot_supervised.ckpt` for the experiments in our paper, and provided `snapshot_self_supervised.ckpt` as an alternative for $\boldsymbol{f_{\phi}}$, as both checkpoints were available to us from other projects at the time. We encourage the research community to try different pre-trained models as the $\boldsymbol{f_{\phi}}$ network.

Regarding the majority-vote accuracy not improving from `snapshot_self_supervised.ckpt`, there are multiple potential explanations: the hyperparameter settings for the training and the diffusion model would have an impact on the result; the pre-trained `snapshot_self_supervised.ckpt` network isn't a good network parameter initialization, and/or doesn't provide good image embeddings; etc. We didn't conduct extensive experiments to build upon this version of $\boldsymbol{f_{\phi}}$, thus couldn't pinpoint the exact reason, but are positive on the improvements if more experiments are performed.

Meanwhile, we'd like to point out that the classification portion of CARD aims to propose the framework of applying the stochasticity of the generative models to make predictions ***with instance level confidence measurements***, thus it is encouraged to put more emphasis on checking the $t$-test results and PIW comparisons besides examining the mean accuracy (the columns to the right of **Accuracy** in Table $5$ of the paper for the CIFAR-10 dataset).
